pkg/artifactory/resource/security/resource_artifactory_distribution_public_key.go

```diff
@@ -0,0 +1,150 @@
+			Required:     true,
+			Description:  "Will be used as a identifier when uploading/retrieving the public key via REST API.",
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"fingerprint": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Returns the computed key fingerprint",
+		},
+		"public_key": {
+			Type:             schema.TypeString,
+			Required:         true,
+			Description:      "The Public key to add as trusted distribution GPG key.",
+			ForceNew:         true,
+			StateFunc:        stripTabs,
+			ValidateDiagFunc: validatePublicKey,
+		},
+		"issued_on": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Returns the date/time when this GPG key was created.",
+		},
+		"issued_by": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Returns the name and eMail address of issuer.",
+		},
+		"valid_until": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Returns the date/time until this GPG key is valid for.",
+		},
+	}
+
+	var resultPacker = packer.Universal(predicate.SchemaHasKey(distributionPublicKeySchema))
+
+	var resourceDistributionPublicKeyCreate = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+		result := distributionPublicKeyPayLoad{}
+
+		resp, err := m.(util.ProvderMetadata).Client.R().SetBody(keyPost{
+			d.Get("alias").(string),
+			stripTabs(d.Get("public_key").(string)),
+		}).SetResult(&result).Post(DistributionPublicKeysAPIEndPoint)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if resp.IsError() {
+			return diag.FromErr(fmt.Errorf("unable to add key: http request failed: %s", resp.Status()))
+		}
+
+		d.SetId(result.KeyID)
+
+		return diag.FromErr(resultPacker(&result, d))
+	}
+
+	var resourceDistributionPublicKeyRead = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+		data := DistributionPublicKeysList{}
+		resp, err := m.(util.ProvderMetadata).Client.R().SetResult(&data).Get(DistributionPublicKeysAPIEndPoint)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if resp.IsError() {
+			return diag.FromErr(fmt.Errorf("unable to read key: http request failed: %s", resp.Status()))
+		}
+
+		for _, key := range data.Keys {
+			if key.KeyID == d.Id() {
+				return diag.FromErr(resultPacker(&key, d))
+			}
+		}
+
+		// If the ID is updated to blank, this tells Terraform the resource no longer exist
+		d.SetId("")
+		return nil
+	}
+
+	var resourceDistributionPublictedKeyDelete = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		resp, err := m.(util.ProvderMetadata).Client.R().Delete(fmt.Sprintf("%s/%s", DistributionPublicKeysAPIEndPoint, d.Id()))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		if resp.IsError() {
+			return diag.FromErr(fmt.Errorf("unable to delete key: http request failed: %s", resp.Status()))
+		}
+
+		return nil
```

Comment:
Pretty sure you need to reset the resource ID manually. At least we used to need to with other resources. e.g. `d.SetId("")`

Reply:
should be fixed in latest commit